pkg/inject/config_override.go

```diff
@@ -0,0 +1,345 @@
+package inject
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	l5dcharts "github.com/linkerd/linkerd2/pkg/charts/linkerd2"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/linkerd/linkerd2/pkg/util"
+	log "github.com/sirupsen/logrus"
+	k8sResource "k8s.io/apimachinery/pkg/api/resource"
+)
+
+type ConfigLookup interface {
+	GetConfigOverride(string) (string, bool)
+	GetInboundPorts() string
+	ParseOpaqueOverrides(string) []util.PortRange
```

Comment:
As far as I can tell, this only depends on the container port values and their names.  Therefore this could be made into a freestanding function which takes a `map[string]int` of port names to numbers.  Callers would need to first create this map from their slice of container specs (or equivalent).

Reply:
good point.